### PR TITLE
Simplify zos-lib contracts

### LINKED ISSUE #88: Simplify backing architecture of contracts for an upgradeable App
Currently we have a huge architecture of smart contracts in order to reflect the different versions of an application of smart contracts on-chain. The idea of this issue is to propose a new simpler model:

For instance, we are providing different flavors of `App`s (versioned and unversioned) when we are only providing support to work with the versioned one through the CLI. We can remove the unversioned one in favour of having an off-chain representation of the smart contracts of your application, which is very much aligned with the lightweight mode we are working on as well (https://github.com/zeppelinos/zos/issues/74).

This will basically trigger the idea of having a single `App` that could have many `Packages`, and then every `Package` can have many `ImplementationsDirectories` as we are doing now.

### PR DESCRIPTION
**Pull request by spalladino**
- Remove VersionedApp/UnversionedApp distinction
- Remove FreezableImplementationDirectory/ImplementationDirectory

Fixes #88

### CODE REVIEW

File: packages/lib/contracts/application/App.sol [hunk 1]
```diff
@@ -1,32 +1,92 @@
...
+  /**
+   * @dev Sets a package in a specific version as a dependency for this application. 
+   * Requires the version to be present in the package.
+   * @param packageName Name of the package to set or overwrite.
+   * @param package Address of the package to register.
+   * @param version Version of the package to use in this application.
+   */
+  function setPackage(string packageName, Package package, string version) public onlyOwner {
+    require(package.hasVersion(version), "The requested version must be registered in the given package");
+    providers[packageName] = ProviderInfo(package, version);
```
Comment (facuspagnuolo) [inline packages/lib/contracts/application/App.sol]: should we check that there was no other package already set under that name?

Comment (facuspagnuolo) [reply packages/lib/contracts/application/App.sol]: I'm thinking if we should force the user to unset the previous package in that case

File: packages/lib/test/contracts/application/App.test.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  describe('ownership', function () {
+    beforeEach("setting ownable", function () {
+      this.ownable = this.app
+    })
+    shouldBehaveLikeOwnable(appOwner, anotherAccount)
+  })
+
+  shouldManageProxies(accounts);
+  shouldManagePackages(accounts);
```
Comment (facuspagnuolo) [inline packages/lib/test/contracts/application/App.test.js]: 👍

File: packages/lib/contracts/application/App.sol (hunk 1 above)
Comment (spalladino) [reply packages/lib/contracts/application/App.sol]: I wouldn't, as this allows the user to use this method to also change the version of a package if they want to do so. And I'd reduce the number of txs to interact with the App as much as possible.

**Pull request merged**